app/scripts/controllers/replicaSet.js

```diff
@@ -205,8 +205,17 @@ angular.module('openshiftConsole')
             errorNotification: false
           }).then(
             // success
-            function(deploymentConfig) {
-              $scope.deploymentConfig = deploymentConfig;
+            function() {
+              watches.push(DataService.watchObject("deploymentconfigs", annotation($scope.replicaSet, 'deploymentConfig'), context, function(deploymentConfig, action) {
```

Comment:
I don't feel like its worth adding a watch on deploymentconfigs just to watch if the DC might get deleted, its an edge case.  It would be different if we were already watching it for something else